app/views/order_items/_order_item.html.erb

```diff
@@ -1,5 +1,5 @@
 <div class="col-md-3 form_products">
-  <div class="thumbnail ">
+  <div class="thumbnail<%= product.stock ==0 ? ' out-of-stock' : '' %>">
```

Comment:
`.nil?` doesn't return true for 0. `zero?` does.

I don't think you need a ternary operator here. `<%= ' out-of-stock' if product.stock.zero? %>` should work too.